libs/wallet/src/use-injected-connector.ts

```diff
@@ -11,28 +17,42 @@ export enum Status {
   AcknowledgeNeeded = 'AcknowledgeNeeded',
 }
 
+export const ERR_VEGA_UNDEFINED = new Error('window.vega not found');
+export const ERR_INVALID_CHAIN = new Error('Invalid chain');
+
 export const useInjectedConnector = (onConnect: () => void) => {
   const { connect, acknowledgeNeeded } = useVegaWallet();
   const [status, setStatus] = useState(Status.Idle);
   const [error, setError] = useState<Error | null>(null);
+  const { VEGA_URL } = useEnvironment();
 
   const attemptConnect = useCallback(
-    async (connector: InjectedConnector, appChainId: string) => {
+    async (
+      connector: InjectedConnector | SnapConnector,
```

Comment:
Would it be simpler to leave useInjectedConnector and create a fresh useSnapConnector hook?